pwndbg/heap/ptmalloc.py

```diff
@@ -547,11 +487,906 @@ def largebin_index(self, sz):
+                    while mp_mmap_threshold_ref.next - mp_ref.address > 0x10:
+                        mp_mmap_threshold_ref = mp_ref
+                        mp_ref = next(iter_possible_match)
+                    self._mp_addr = base_offset + mp_ref.disp
+                except StopIteration:
+                    pass
+            elif pwndbg.arch.current == "aarch64":
+                base_offset = pwndbg.vmmap.find(pwndbg.symbol.address('_IO_list_all')).start
+                regs = set()
+                found = False
+                for instr in __libc_free_instructions:
+                    if found:
+                        break
+                    # We want to find sth like: `str reg2, [reg1, #offset]``
+                    # and reg1 is from `adrp reg1, base_offset`
+                    # We can notice that it only have one match in __libc_free
+                    # The match should be the reference to mp_
+                    if instr.mnemonic == 'str':
+                        for reg in regs:
+                            if "[" + reg in instr.op_str:
+                                self._mp_addr = base_offset + instr.operands[1].mem.disp
+                                found = True
+                                break
+                    elif instr.mnemonic == 'adrp' and instr.operands[1].int == base_offset:
+                        regs.add(instr.operands[0].str)
+            elif pwndbg.arch.current == "arm":
+                regs = {}
+                ldr = {}
+                found = False
+                for instr in __libc_free_instructions:
+                    if found:
+                        break
+                    # We want to find sth like: `str reg2, [reg1, #8 or #0]`
+                    # and reg1 is from `ldr reg1, [pc, #offset]` and `add reg1, pc`
+                    # We can notice that it only have one match in __libc_free
+                    # The match should be the reference to mp_ and mp_.mmap_threshold
+                    if instr.mnemonic == 'str':
+                        for reg in regs:
+                            if "[" + reg + "]" in instr.op_str:
+                                # ldr reg1, [pc, #offset]
+                                offset = regs[reg].operands[1].mem.disp
+                                offset = pwndbg.memory.s32((regs[reg].address + 4 & -4) + offset)
+                                # add reg1, pc
+                                self._mp_addr = offset + ldr[reg].address + 4
+                                found = True
+                                break
+                    elif instr.mnemonic == 'add':
+                        for reg in regs:
+                            if instr.op_str == reg + ", pc":
+                                ldr[reg] = instr
+                    elif instr.mnemonic == 'ldr' and "[pc," in instr.op_str:
+                        regs[instr.operands[0].str] = instr
+            else:
+                inform_report_issue("mp_")
```

Comment:
Do we need this else here if there is another one below?

Reply:
Maybe don't, so more architectures might be supported by heuristic.
Thanks! I'll fix it.